AutomatedTesting/Gem/PythonTests/AWS/common/constants.py

```diff
@@ -6,11 +6,12 @@
 """
 
 import os
+import platform
 
 # ARN of the IAM role to assume for retrieving temporary AWS credentials
 ASSUME_ROLE_ARN = os.environ.get('ASSUME_ROLE_ARN', 'arn:aws:iam::645075835648:role/o3de-automation-tests')
 # Name of the AWS project deployed by the CDK applications
-AWS_PROJECT_NAME = os.environ.get('O3DE_AWS_PROJECT_NAME', 'AWSAUTO')
+AWS_PROJECT_NAME = (os.environ.get('BRANCH_NAME', '') + '-' + os.environ.get('PIPELINE_NAME', '') + '-' + platform.system()).upper()

```

Comment:
minor: Danger here is that you will blow out resource names, they will either be too long or all have BRANCH-PIPELINE and get truncated at the important part of the resource name.

More a concern to be careful about rather than actual issue.

Reply:
I believe that we addressed some of the problems in our own CDK applications. I've tested with a long name locally and the tests passed. Should be fine for the currently Jenkins pipeline setup.

Reply:
Minor: what is the reason behind changing project name to include branch and pipeline name? as we have removed O3DE_AWS_PROJECT_NAME from the jenkins pipeline parameters, then why not just make project name simple? (considering stack gets delete after test)
if so, then you don't even need to use BRANCH_NAME and PIPELINE_NAME these two parameters as well.